src/components/AuthenticationPage.js

```diff
@@ -2,29 +2,50 @@ import Greeting from "./Greeting";
 import Header from "./Header";
 import AuthenticationForm from "./AuthenticationForm";
 
+const signup = {
+  title: "Sign Up",
+  linkMessage: "Have an account?",
+  link: "#",
+  placeholders: ["Username", "Email", "Password"],
+  name: ["username","email","password"],
+  inputType: ["text", "text", "password"],
+  buttonMessage: "Submit"
+}
+const signin = {
+  title: "Sign In",
+  linkMessage: "Need an account?",
+  link: "#",
+  placeholders: ["Email", "Password"],
+  name: ["email","password"],
+  inputType: ["text", "text"],
+  buttonMessage: "Sign in"
+}
+const userData = {}
+
 function AuthenticationPage({isSignin}){
-  const signup = {
-    title: "Sign Up",
-    linkMessage: "Have an account?",
-    link: "#",
-    placeholders: ["Username", "Email", "Password"],
-    buttonMessage: "Submit"
-  }
-  const signin = {
-    title: "Sign In",
-    linkMessage: "Need an account?",
-    link: "#",
-    placeholders: ["Email", "Password"],
-    buttonMessage: "Sign in"
-  }
   return (
     <div>
       <Header />
       <div className="flex flex-col justify-center items-center">
         <Greeting title={ isSignin ? signin.title : signup.title } linkMessage={ isSignin ? signin.linkMessage : signup.linkMessage } link={ isSignin ? signin.link : signup.link } />
-        <AuthenticationForm placeholders={ isSignin ? signin.placeholders : signup.placeholders } buttonMessage={ isSignin ? signin.buttonMessage : signup.buttonMessage } />
+        <AuthenticationForm submitCallbackFn={ isSignin ? handleSubmitSignIn : handleSubmitSignUp } inputType={isSignin ? signin.inputType : signup.inputType }  placeholders={ isSignin ? signin.placeholders : signup.placeholders } name={ isSignin ? signin.name : signup.name } buttonMessage={ isSignin ? signin.buttonMessage : signup.buttonMessage } />
       </div>
     </div>
   )
 }
+
+function handleSubmitSignIn(){
+}
+
+function handleSubmitSignUp(event){
+  event.preventDefault();
+  signup.name.forEach((inputName) => {
+    const input = event.target[inputName];
+    userData[inputName] = input.value;
+
+    input.value = "";
+  })
+  sessionStorage.setItem("user", JSON.stringify(userData));
```

Comment:
Don't save the entire response to the session storage. Only the JWT token since that is the only thing that is required to verify if signed in or not. In a future PR we can save fetch the user data and save it in Redux.

Reply:
I refactored this statement and prepared this part for the endpoint that receives the token.